src/containers/DownloadContainer/DownloadContainer.jsx

```diff
@@ -0,0 +1,120 @@
+import React, { useEffect, useState } from 'react'
+import { FlatList, View } from 'react-native'
+import PropTypes from 'prop-types'
+import * as FileSystem from 'expo-file-system'
+import { Icon, Image, ListItem } from 'react-native-elements'
+import * as VideoThumbnails from 'expo-video-thumbnails'
+import { STACK_SCREENS } from '../CTNavigationContainer/index'
+
+/**
+ * Contains the home screen of the application. Lists starred courses and gives the user the ability
+ * to search for courses. Clicking on a course shows the playlists for it.
+ */
+const DownloadContainer = ({ navigation }) => {
+  const [localVideos, setLocalVideos] = useState([]) // an array of local video names
+
+  const getFileSize = async (fileUri) => {
+    const fileInfo = await FileSystem.getInfoAsync(fileUri)
+    return (fileInfo.size / (1024 * 1024)).toFixed(2)
+  }
+
+  const generateThumbnail = async (videoPath) => {
+    try {
+      const { uri } = await VideoThumbnails.getThumbnailAsync(videoPath, {
+        time: 15000,
+      })
+      return uri
+    } catch (e) {
+      // eslint-disable-next-line no-console
+      console.warn(e)
+      return null
+    }
+  }
+
+  const fetchLocalFiles = async () => {
+    let files = await FileSystem.readDirectoryAsync(FileSystem.documentDirectory)
+    files = files.filter((fileName) => fileName.endsWith('.mp4'))
```

Comment:
Will this grab all files that end with mp4? Even ones not related to CT

Reply:
`FileSystem.documentDirectory` is currently only used for storing downloaded videos. So we would not worry about this problem unless we decided to download other documents.

Reply:
I guess I was more so asking if other applications can download to this directory. I.e. if I download a movie with something like netflix, is there a chance that we might accidentally pull that from the file?

Reply:
No, the expo filesystem would assign each app a unique document folder and this folder would only be accessed by this app.